test/path_tokenizer.h

```diff
@@ -0,0 +1,56 @@
+//
+// Created by wkl04 on 12-12-2019.
+//
+
+#ifndef MQTTSUBSCRIPTION_PATH_TOKENIZER_H
+#define MQTTSUBSCRIPTION_PATH_TOKENIZER_H
+
+#include <algorithm>
+
+template<typename Output>
+inline void mqtt_path_tokenizer(MQTT_NS::string_view str, Output write)
```

Comment:
I feel like this should be abstracted more.

Why not create two overloads of this function.

1.
```cpp
template<typename Begin, typename End, typename Output>
inline void mqtt_path_tokenizer(Begin first, End const& last, Output write)
```

then

2. 
```cpp
template<typename Str, typename Output>
inline void mqtt_path_tokenizer(Str const& str, Output && write) {
    mqtt_path_tokenizer(std::begin(str), std::end(str), std::forward<Output>(write));
}
```